fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBDatabaseFactory.java

```diff
@@ -366,6 +378,26 @@ public synchronized void setRunLoopProfilingEnabled(boolean runLoopProfilingEnab
         this.runLoopProfilingEnabled = runLoopProfilingEnabled;
     }
 
+    /**
+     * Set the number of threads per FDB client version. The default value is 1.
+     *
+     * @param tPerClientV the number of threads per client version. Cannot be less than 1
+     */
+    public static synchronized void setThreadsPerClientVersion(int tPerClientV) {
+        if (staticOptionsSet) {
+            throw new RecordCoreException("threads per client version cannot be changed as the version has already been initiated");
+        }
+        if (tPerClientV < 1) {
+            //if the thread count is too low, disable the setting
+            tPerClientV = 1;
+        }
+        threadsPerClientVersion = tPerClientV;
+    }
+
+    public synchronized int getThreadsPerClientVersion(int threadsPerClientVersion) {
```

Comment:
```suggestion
    public synchronized int getThreadsPerClientVersion() {
```

I'm also not sure we want/need this to be synchronized, given that it's a read-only operation

Reply:
if `threadsPerClientVersion` isn't synchronized, then this has to be (otherwise you might never actually see the changed number)

Reply:
I am not really _worried_, but it's good design to make sure that your concurrency model works. I too prefer `volatile`, but was attempting to be responsive to earlier comments.

I reverted to `volatile` and further clarified the concurrency model.